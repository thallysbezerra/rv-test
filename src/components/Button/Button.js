import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = ({ children, className, icon, iconTitle, onClick, small }) => {
	return (
		<button
			className={`button${small ? '--small' : ''} ${
				className ? className : ''
				}`}
			onClick={onClick}
		>
			{children}
			{icon && <img src={icon} alt={iconTitle} className="button__icon" />}
		</button>
	);
};

Button.propTypes = {
	children: PropTypes.string.isRequired,
	className: PropTypes.string.isRequired,
	icon: PropTypes.string,
	iconTitle: PropTypes.string,
	onClick: PropTypes.func.isRequired,
	small: PropTypes.bool
};

Button.defaultProps = {
	children: "Button name",
	className: "",
	onClick: () => { }
}

export default Button;
