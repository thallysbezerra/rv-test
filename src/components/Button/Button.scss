@import '../../css/theme.scss';

.button {
	align-items: center;
	background-color: transparent;
	border: none;
	display: flex;
	height: 35px;

	&__icon {
		margin-left: 10px;
	}

	&--small {
		@extend .button;

		letter-spacing: 9.5px;
		text-transform: uppercase;
	}

	&:hover {
		cursor: pointer;
	}
}
