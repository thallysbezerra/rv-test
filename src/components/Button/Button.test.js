import React from 'react';
import Button from './Button';

import renderer from 'react-test-renderer';

describe('Button UI', () => {
	it('<Button /> must have the snapshot version.', () => {
		const tree = renderer.create(<Button />);

		expect(tree.toJSON()).toMatchSnapshot();
	});
});
