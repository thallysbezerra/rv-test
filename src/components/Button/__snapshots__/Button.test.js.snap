// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI <Button /> must have the snapshot version. 1`] = `
<button
  className="button "
  onClick={[Function]}
>
  Button name
</button>
`;
